packages/botonic-plugin-directus/src/cms/contents.ts

```diff
@@ -0,0 +1,50 @@
+export interface CommonFields {
+  id: string
+  name: string
+  shortText?: string
+  keywords?: string[]
+  followup?: Text | Image
+  customFields?: Record<string, unknown>
+}
+
+export abstract class Content {
+  readonly common: CommonFields
+  constructor(common: CommonFields) {
+    this.common = common
+  }
+  get name(): string {
+    return this.common.name
+  }
+
+  get id(): string {
+    return this.common.id
+  }
+}
+
+export class Text extends Content {
+  readonly text: string
+  readonly buttons?: Button[]
+  constructor(opt: { common: CommonFields; text: string; buttons?: Button[] }) {
```

Comment:
I would define specific interfaces for the fields of every content, for example:
```
export interface TextFields {
    common: CommonFields
    text: string
    buttons?: Button[]
}
```
And then the constructor would be like this:
```
constructor(opt: TextFields) {
```